examples/tp/tm_high_order.py

```diff
@@ -0,0 +1,465 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+A simple tutorial that shows some features of the Temporal Memory.
+
+The following program has the purpose of presenting some
+basic properties of the Temporal Memory, in particular when it comes
+to how it handles high-order sequences.
+"""
+
+import numpy as np
+import random
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+from nupic.bindings.algorithms import TemporalMemory as TM
+
+def accuracy(curr, pred):
+  """
+  Computes the accuracy of the TM at time-step t based on the prediction
+  at time-step t-1 and the current active columns at time-step t.
+  
+  @param curr (array) binary vector containing current active columns
+  @param pred (array) binary vector containing predicted active columns
+  
+  @return acc (float) prediction accuracy of the TM at time-step t
+  """  
+  acc = 0
+  if np.count_nonzero(pred) > 0:
+    acc = float(np.dot(curr, pred))/float(np.count_nonzero(pred))
+  return acc
+
+def corruptVector(v1, noiseLevel, numActiveCols):
+  """
+  Corrupts a copy of a binary vector by inverting noiseLevel percent of its bits.
+  
+  @param v1      (array) binary vector whose copy will be corrupted
+  @param noiseLevel  (float) amount of noise to be applied on the new vector
+  @param numActiveCols (int)   number of sparse columns that represent an input
+  
+  @return v2 (array) corrupted binary vector
+  """  
+  size = len(v1)
+  v2 = np.zeros(size, dtype="uint32")
+  bitsToSwap = int(noiseLevel * numActiveCols)
+  # Copy the contents of v1 into v2
+  for i in range(size):
+    v2[i] = v1[i]
+  for _ in range(bitsToSwap):
+    i = random.randrange(size)
+    if v2[i] == 1:
+      v2[i] = 0
+    else:
+      v2[i] = 1
+  return v2
+
+def showPredictions():
+  """
+  Shows predictions of the TM when presented with the characters A, B, C, D, X, and
+  Y without any contextual information, that is, not in a sequence.
```

Comment:
The comment is a little unclear. You are showing ABCDXY in a sequence. Did you mean to do a reset in between each element? Or do you mean that there is no prior context before showing the sequence? I think you might mean the latter. In that case you should do a tm.reset before the for loop to guarantee there is no prior context.